File: backend/utils/Logger.ts
export default class Logger {
  // Info, Warning, Error, Debug, Success
  private static colors = {
    info: '\x1b[36m%s\x1b[0m',
    warn: '\x1b[33m%s\x1b[0m',
    error: '\x1b[31m%s\x1b[0m',
    debug: '\x1b[34m%s\x1b[0m',
    success: '\x1b[32m%s\x1b[0m',
    reset: '\x1b[0m',
  };

  public static info(message: string, details?: string): void {
    console.log(Logger.colors.info, message, Logger.colors.reset, details ? details : '');
  }

  public static warn(message: string, details?: string): void {
    console.log(Logger.colors.warn, message, Logger.colors.reset, details ? details : '');
  }

  public static error(message: string, details?: string): void {
    console.log(Logger.colors.error, message, Logger.colors.reset, details ? details : '');
  }

  public static debug(message: string, details?: string): void {
    console.log(Logger.colors.debug, message, Logger.colors.reset, details ? details : '');
  }

  public static success(message: string, details?: string): void {
    console.log(Logger.colors.success, message, Logger.colors.reset, details ? details : '');
  }
}

File: backend/pkg/logger/logger.go
package logger

import (
	"fmt"
	"log"
	"os"
)

const (
	reset  = "\033[0m"
	green  = "\033[32m"
	red    = "\033[31m"
	yellow = "\033[33m"
	blue   = "\033[36m"
)

var useColor = true

var baseLogger = log.New(os.Stdout, "", 0)

func format(level string, color string, msg string) string {

	if useColor {
		return fmt.Sprintf("%s%s%s %s", color, level, reset, msg)
	}

	return fmt.Sprintf("%s %s", level, msg)
}

func Info(title, msg string) {
	baseLogger.Println(format(title+" |", blue, msg))
}

func Warn(title, msg string) {
	baseLogger.Println(format(title+" |", yellow, msg))
}

func Error(title, msg string) {
	baseLogger.Println(format(title+" |", red, msg))
}

func Success(title, msg string) {
	baseLogger.Println(format(title+" |", green, msg))
}

// Optionnel : activer/désactiver les couleurs globalement
func UseColor(b bool) {
	useColor = b
}
